In [1]:
import pandas as pd
import numpy as np


In [2]:
play = pd.read_csv('googleplaystore.csv')

In [3]:
#Checking data
play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [4]:
play.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [5]:
play = play.rename(columns = {'Content Rating': 'Content_Rating', 'Last Updated': 'Last_Updated', 'Current Ver': 'Current_Ver', 'Android Ver': 'Android_Ver'})

In [6]:
play[play['App']== 'Life Made WI-Fi Touchscreen Photo Frame']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content_Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last_Updated Current_Ver Android_Ver  
10472       1.0.19  4.0 and up         NaN

In [7]:
play['Current_Ver'] = play['Current_Ver'].str.replace('4.0 and up', '1.0.19')
play['Last_Updated'] = play['Last_Updated'].str.replace('1.0.19', 'February 11, 2018')
play['Genres'] = play['Genres'].str.replace('February 11, 2018', 'Lifestyle')
play['Content_Rating'] = play['Content_Rating'].fillna('Everyone')
play['Price'] = play['Price'].str.replace('Everyone', 'Free')
play['Type'] = play['Type'].str.replace('0', 'Free')
play['Installs'] = play['Installs'].str.replace('Free', '1000')
play['Size'] = play['Size'].str.replace('1,000+', '3')
play['Size'] = play['Size'].str.replace('+', '')
play['Reviews'] = play['Reviews'].str.replace('3.0M', '19')
play['Rating'] = play['Rating'].replace(19,1.9)
play['Category'] = play['Category'].str.replace('1.9', 'LIFESTYLE')
play[play['App']== 'Life Made WI-Fi Touchscreen Photo Frame']


App   Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame  LIFESTYLE     1.9      19   

      Size Installs  Type Price Content_Rating     Genres       Last_Updated  \
10472    3     1000  Free  Free       Everyone  Lifestyle  February 11, 2018   

      Current_Ver Android_Ver  
10472      1.0.19         NaN

Define:
Several of the columns have values that are not required. The + sign from Installs, the 'and up' from Android Ver, the 'M' from Size. This section makes those changes

In [10]:
#Removing the Plus sign from installs
play['Installs'] = play['Installs'].map(lambda x: x.rstrip('+')).str.replace(',','')

#Removing the 'and up' from Android Version
play['Android_Ver'] = play['Android_Ver'].str.replace('and up','')

play['Size'] = play['Size'].str.replace('M','')

play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews Size  Installs  Type Price Content_Rating  \
0     159   19     10000  Free     0       Everyone   
1     967   14    500000  Free     0       Everyone   
2   87510  8.7   5000000  Free     0       Everyone   
3  215644   25  50000000  Free     0           Teen   
4     967  2.8    100000  Free     0       Everyone   

                      Genres      Last_Updated         Current_Ver Android_Ver  
0               Art & Design   January 7, 2018               1.0.0      4.0.3   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0      4.0.3   
2               Art & Design    August 1, 2018               1.2.4      4.0.3   
3               Art & Design      June 8, 2018  Varies with device        4.2   
4    Art & Design;Creativity     June 20, 2018                 1.1        4.4

Rewviews, size, installs, price need to change to floats(int)

In [11]:
#Correcting the Life made error
play['Installs'] = play['Installs'].str.replace('Free','1000')
play['Installs'] = play.Installs.astype('int32')

Test

In [ ]:
play.head()

DEFINE:
Android Ver needs to removed and up

Code

In [ ]:
play

Total rows 10841


Installs has plus symbol that needs to be removed

Android ver needs to remove and up

M removed from size

Rewviews, size, installs, price need to change to floats(int)
Last updated changed to datetime

Remove _ from Category

Change catrgory to Upper-lowercase

Possibly split up Genres for games with multiple genres

App Names as Uppercase for each level

CC Rivals, Type needs to be changed to Free

Life Made WI-Fi Touchscreen changed to Everyone

Current Version Changes
1.1.2 - Learn To Draw Kawaii Characters
4 - Virtual DJ Sound Mixer
1.1.2 - BT Master	
2.17.8 - Dots puzzle


Life Made WI-Fi Touchscreen Photo Frame Cat change to Lifestyle, everything has shifted over

### Category

In [ ]:
cat = play.groupby('Installs')['App'].count().sort_values()
cat